### PR TITLE
Add Amezmo Drupal settings

### PR DESCRIPTION
**Pull request by verbeeksteven**
- Moves settings to individual file and looks for `APP_HOSTNAME` to setup
- Add after.pull to link `/web/sites/default/files` to `/webroot/storage/files` and make it if it doesn't exist
- Setup redis settings
- Autoload dotenv settings using composer

TODO:

- Figure out how to map mysql hostname and user to ENV variables
- Figure out how to get APP_KEY (or some encrypted key that can be used for salt)

### CODE REVIEW

File: web/sites/default/settings.amezmo.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * @file
+ * Platform.sh settings.
+ */
+
+// Configure the database.
+if (isset($_ENV['APP_HOSTNAME'])) {
+  // TODO: FIND A WAY TO GET DB_HOST AND DB_USER
+  $databases['default']['default'] = [
```
Comment (rmccullagh) [inline web/sites/default/settings.amezmo.php:10]: We can generate APP_HOSTNAME as a Drupal specific configuration that would be autogenerated by Amezmo, so I agree with this.

File: web/sites/default/settings.amezmo.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    'database' => $_ENV['DB_DATABASE'],
```
Comment (rmccullagh) [inline web/sites/default/settings.amezmo.php:11]: For Drupal apps, `MYSQL_DATABASE`, `MYSQL_HOSTNAME`, `MYSQL_PASSWORD`, and `MYSQL_USER` can be used here which are generated in the .env file that Amezmo generates.

File: web/sites/default/settings.amezmo.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  $settings['php_storage']['default']['directory'] = $settings['file_private_path'];
+  $settings['php_storage']['twig']['directory'] = $settings['file_private_path'];
+
+  $settings['file_chmod_directory'] = 2775;
+  $settings['file_chmod_file'] = 0664;
+
+  // Set the project-specific entropy value, used for generating one-time
+  // keys and such.
+  // TODO: FIND A WAY TO REPLACE THIS WITH APP_KEY
+  $settings['hash_salt'] = '';
```
Comment (rmccullagh) [inline web/sites/default/settings.amezmo.php:33]: Does Drupal have a recommended hash_salt format? If so we can generate this automatically or use something similar to Laravel's `APP_KEY`

**Review by rmccullagh — APPROVED** [review]
Looks great. Just one thing, if we want to use the DB_* pattern, that is fine. Currently Amezmo generates MYSQL_* constants, but I think it would be better for them to be generic as in DB_* - We will change the auto generated .env file based on this PR.

File: web/sites/default/settings.amezmo.php (hunk 3 above)
Comment (verbeeksteven) [reply web/sites/default/settings.amezmo.php:33]: It can be anything, as long as it's something. So what ever you use for Laravel or CraftCMS works.

**Pull request merged**